pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -680,6 +710,20 @@ func (s *Service) LaunchTemplateNeedsUpdate(scope scope.LaunchTemplateScope, inc
 	if incoming.InstanceType != existing.InstanceType {
 		return true, nil
 	}
+	if !cmp.Equal(incoming.InstanceMetadataOptions, existing.InstanceMetadataOptions) {
+		return true, nil
+	}
+
+	fmt.Println("STEFANCMP", incoming.InstanceMetadataOptions, existing.InstanceMetadataOptions)
+
+	if !cmp.Equal(incoming.InstanceMetadataOptions, existing.InstanceMetadataOptions) {
```

Comment:
Wouldn't compare already compete these fields?

Reply:
Rebasing after they were out of sync added the above back in. Ill fix that now.